### PR TITLE
fixes #4685 feat(project): only put accepted experiments into draft on rejection

### LINKED ISSUE #4685: Handle RS rejections gracefully for all possible rejection cases
There are three times we make a review request to Remote Settings:

- On launch
- On pause
- On end

We currently handle the case where a launch is rejected and capture the rejection reason, but we don't handle the case where it's rejected on pause/end.

I think we should

1) Filter the launch rejection case to make sure we only attempt to roll back an experiment that's in the accepted state
2) In all other cases, throw out the rejection reason and put the collection back in the rollback state so that the pause/end tasks will just attempt the change again.  So then if someone rejects a pause/end (which should not be part of the workflow), the request will just come back on the next invocation.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-999)
┆Issue Number: EXP-999

### PR DESCRIPTION
**Pull request by jodyheavener**
Closes #4685

This work is slightly different from the issue description per my discussion with @jaredlockhart. This PR updates `handle_rejection` (kicked off by the `nimbus_check_kinto_push_queue` task) to:

- Add a condition to the elif that changes the experiment's status back to Draft only if the experiment was in an Accepted state to begin with
- Conditionally saves the experiment in the DB to avoid unnecessary writes

As we discussed, any reject will still 1) create a changelog, and 2) trigger a rollback

### CODE REVIEW

File: app/experimenter/kinto/tasks.py
Comment (jaredlockhart) [inline app/experimenter/kinto/tasks.py:104]: 👍

**Review by jaredlockhart — COMMENTED** [review]
Wait I'm confused, this is PRing into your other branch and not into main?

**Comment by jodyheavener** [comment]
@jaredlockhart yeah because the first one isn’t merged and relies on it, this just keeps those commits out of this PR

**Comment by jaredlockhart** [comment]
But why not just make this change against main and then rebase your other pr on top of it?

**Comment by jodyheavener** [comment]
Wouldn't that add those commits to this PR because I'd be comparing against main? idk I've just always done it this way, don't see the harm?

**Comment by jodyheavener** [comment]
So I chnanged it back to main, but because main doesn't have those commits they get included here and I wouldn't want this PR to merge them?

**Comment by jaredlockhart** [comment]
No I mean make this change in isolation against main with 1 commit with the rejection logic where it is, then rebase and incorporate it into your other PR.  With continuous delivery every PR needs to be isolated and revertable, so if you PR this into your other PR and then we merge that into main and have to revert it we lose both changes.

**Comment by jaredlockhart** [comment]
For reference: https://github.com/mozilla/experimenter/blob/main/contributing.md#continuous-deployment-process

**Comment by jodyheavener** [comment]
Oh I see - I wasn't going to merge it into the other PR branch, I was going to wait until that branch was merged and then change this base branch back to main. Since this PR is not urgent it's just a shortcut to avoiding potential conflicts.

**Comment by jaredlockhart** [comment]
Okay sure then just move this into a draft and we can merge the other one first?

**Comment by jodyheavener** [comment]
That's fine, I just wanted to signify that the code technically is ready for review and I don't think you can r+ if it's in Draft.

**Comment by jaredlockhart** [comment]
Yeah I'd rather wait until things are actually landable before approving

**Review by jaredlockhart — APPROVED** [review]
Works perfectly tytyty 👏 👏 👏

**Pull request merged**
